### PR TITLE
Refactor code, remove warnings

### PR DESCRIPTION
**Pull request by sendyhalim**
Refactor code to remove xcode warnings

### CODE REVIEW

File: Source/Moya+Internal.swift [hunk 1]
```diff
@@ -203,7 +203,7 @@ private extension MoyaProvider {
         }
 
         // Perform the actual request
-        if let progress = progressCompletion {
+        if progressCompletion != nil {
```
Comment (BasThomas) [inline Source/Moya+Internal.swift]: What about `if let _ = progressCompletion` Instead of the `nil` check?

Comment (ashfurrow) [inline Source/Moya+Internal.swift]: I personally prefer `if let _ = ` as well, wasn't sure if it was just me 😅

Comment (BasThomas) [inline Source/Moya+Internal.swift]: 🙌

Comment (sendyhalim) [inline Source/Moya+Internal.swift]: Sure, that's better! 😄

**Comment by BasThomas** [comment]
I suppose that changelog entry is not needed here. 😉

**Pull request merged**